llvm/lib/Target/ICMC/MCTargetDesc/ICMCMCTargetDesc.cpp

```diff
@@ -24,13 +30,19 @@ ICMCMCAsmInfo::ICMCMCAsmInfo(const Triple &TT, const MCTargetOptions &Options){
 static MCRegisterInfo *createICMCMCRegisterInfo(const Triple &TT) {
     MCRegisterInfo *X = new MCRegisterInfo();
     InitICMCMCRegisterInfo(X, ICMC::SP, 0, 0, ICMC::PC);
-    printf("%p\n", (void*) X);
+    return X;
+}
+
+static MCInstrInfo *createICMCMCInstrInfo(void) {
+    MCInstrInfo *X = new MCInstrInfo();
+    InitICMCMCInstrInfo(X);
     return X;
 }
 
 extern "C" LLVM_EXTERNAL_VISIBILITY void LLVMInitializeICMCTargetMC() {
     Target T = getTheICMCTarget();
     RegisterMCAsmInfo<ICMCMCAsmInfo> X(T);
+    TargetRegistry::RegisterMCInstrInfo(T, createICMCMCInstrInfo);
```

Comment:
Great, the instruction info is already initialized in the same order as the other implementations from what I remember